ext/native/thin3d/thin3d_vulkan.cpp

```diff
@@ -946,24 +957,26 @@ VkDescriptorSet VKContext::GetOrCreateDescriptorSet(VkBuffer buf) {
 		numWrites++;
 	}
 
-	if (key.imageView_ && boundSamplers_[0] && boundSamplers_[0]->GetSampler()) {
-		imageDesc.imageView = key.imageView_ ? key.imageView_ : VK_NULL_HANDLE;
-		imageDesc.sampler = boundSamplers_[0] ? boundSamplers_[0]->GetSampler() : VK_NULL_HANDLE;
+	for (int i = 0; i < MAX_BOUND_TEXTURES; ++i) {
+		if (key.imageViews_[i] && key.samplers_[i] && key.samplers_[i]->GetSampler()) {
+			imageDesc.imageView = key.imageViews_[i];
+			imageDesc.sampler = key.samplers_[i]->GetSampler();
 #ifdef VULKAN_USE_GENERAL_LAYOUT_FOR_COLOR
-		imageDesc.imageLayout = VK_IMAGE_LAYOUT_GENERAL;
+			imageDesc.imageLayout = VK_IMAGE_LAYOUT_GENERAL;
 #else
-		imageDesc.imageLayout = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
+			imageDesc.imageLayout = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
 #endif
-		writes[numWrites].sType = VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET;
-		writes[numWrites].dstSet = descSet;
-		writes[numWrites].dstArrayElement = 0;
-		writes[numWrites].dstBinding = 1;
-		writes[numWrites].pBufferInfo = nullptr;
-		writes[numWrites].pImageInfo = &imageDesc;
-		writes[numWrites].pTexelBufferView = nullptr;
-		writes[numWrites].descriptorCount = 1;
-		writes[numWrites].descriptorType = VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER;
-		numWrites++;
+			writes[numWrites].sType = VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET;
+			writes[numWrites].dstSet = descSet;
+			writes[numWrites].dstArrayElement = 0;
+			writes[numWrites].dstBinding = i + 1;
+			writes[numWrites].pBufferInfo = nullptr;
+			writes[numWrites].pImageInfo = &imageDesc;
```

Comment:
Here the problem, silly me for not seeing it before. You need to make imageDesc an array too to have different one for each texture.

This fix it:
```
diff --git a/ext/native/thin3d/thin3d_vulkan.cpp b/ext/native/thin3d/thin3d_vulkan.cpp
index cdcadbea2..46ed7fe66 100644
--- a/ext/native/thin3d/thin3d_vulkan.cpp
+++ b/ext/native/thin3d/thin3d_vulkan.cpp
@@ -936,7 +936,7 @@ VkDescriptorSet VKContext::GetOrCreateDescriptorSet(VkBuffer buf) {
        bufferDesc.offset = 0;
        bufferDesc.range = curPipeline_->GetUBOSize();
 
-       VkDescriptorImageInfo imageDesc;
+       VkDescriptorImageInfo imageDesc[MAX_BOUND_TEXTURES];
 
        VkWriteDescriptorSet writes[1 + MAX_BOUND_TEXTURES] = {};
 
@@ -959,19 +959,19 @@ VkDescriptorSet VKContext::GetOrCreateDescriptorSet(VkBuffer buf) {
 
        for (int i = 0; i < MAX_BOUND_TEXTURES; ++i) {
                if (key.imageViews_[i] && key.samplers_[i] && key.samplers_[i]->GetSampler()) {
-                       imageDesc.imageView = key.imageViews_[i];
-                       imageDesc.sampler = key.samplers_[i]->GetSampler();
+                       imageDesc[i].imageView = key.imageViews_[i];
+                       imageDesc[i].sampler = key.samplers_[i]->GetSampler();
 #ifdef VULKAN_USE_GENERAL_LAYOUT_FOR_COLOR
-                       imageDesc.imageLayout = VK_IMAGE_LAYOUT_GENERAL;
+                       imageDesc[i].imageLayout = VK_IMAGE_LAYOUT_GENERAL;
 #else
-                       imageDesc.imageLayout = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
+                       imageDesc[i].imageLayout = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
 #endif
                        writes[numWrites].sType = VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET;
                        writes[numWrites].dstSet = descSet;
                        writes[numWrites].dstArrayElement = 0;
                        writes[numWrites].dstBinding = i + 1;
                        writes[numWrites].pBufferInfo = nullptr;
-                       writes[numWrites].pImageInfo = &imageDesc;
+                       writes[numWrites].pImageInfo = &imageDesc[i];
                        writes[numWrites].pTexelBufferView = nullptr;
                        writes[numWrites].descriptorCount = 1;
                        writes[numWrites].descriptorType = VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER;
```

Reply:
Gah, thank you.  Now it looks obvious, silly mistake.  Can't blame the validation layer for this one.

-[Unknown]